source/common/version/version.cc

```diff
@@ -37,9 +37,10 @@ const envoy::config::core::v3::BuildVersion& VersionInfo::buildVersion() {
 }
 
 bool VersionInfo::sslFipsCompliant() {
-  bool fipsCompliant = false;
-#ifdef BORINGSSL_FIPS
-  fipsCompliant = true;
+#ifdef ENVOY_SSL_FIPS
+  static bool fipsCompliant = true;
```

Comment:
Why do we need the static? Why not just
```
#ifdef ENVOY_SSL_FIPS
return true;
#else 
return false;
#endif 
```

Reply:
@jmarantz  Initially I thought this initialization must happen during compile time as "BORINGSSL_FIPS" is being pass during compilation. But it seems like it is not required. Should I change that?

Reply:
Yes, you don't need a static at all for this use-case. Just return the value based on the ifdef.

Reply:
Thanks, I will change it.